server/routes/updateGoal/index.ts

```diff
@@ -26,4 +27,21 @@ export default (router: Router, services: Services) => {
     checkPrisonerSummaryExistsInSession,
     updateGoalController.submitUpdateGoalForm,
   ])
+
+  router.use('/plan/:prisonNumber/goals/:goalReference/update/review', [
+    checkUserHasEditAuthority(),
+    prisonerSummaryRequestHandler.getPrisonerSummary,
+  ])
+
+  router.get('/plan/:prisonNumber/goals/:goalReference/update/review', [
+    checkPrisonerSummaryExistsInSession,
+    checkUpdateGoalFormExistsInSession,
```

Comment:
Good work finding our how our route request handlers work 👍 💯 

What you've done works, and is fine, but there is an improvement I might suggest (which might help solidify how and why some of this stuff works)

`router.use` is like a 'common' handler/hook for the given path that are then referenced in the `router.get` / `router.post` etc calls. It means we can put common code there and not have to repeat it in the `router.get` / `router.post` etc calls.
In `router.use` we want to check the user has the correct authority 👍 , but we can also use it to check the user has the Update Goal Form in the session. It therefore saves doing it in both the `router.get` / `router.post` calls.

And we can improve it even more ... 👍 ....
The call `prisonerSummaryRequestHandler.getPrisonerSummary` in the current `router.use` goes off and puts the prisoner summary in the session if it's not already there. Putting it in a `router.use` makes sense .... but only when its the start of the journey.
Our new page is the end of the journey (its the check your answers page), so we dont want to mess about getting the prisoner summary and putting it in the session for _this_ route - we want to make sure its already there. If not it means someone is trying to get to this page out of sequence.
So, you can remove the call to get the prisoner summary (line 34), and replace it with a check to see that it already exists.

A bit like this: ....

```
  router.use('/plan/:prisonNumber/goals/:goalReference/update/review', [
    checkUserHasEditAuthority(),
    checkPrisonerSummaryExistsInSession,
    checkUpdateGoalFormExistsInSession,
  ])

  router.get('/plan/:prisonNumber/goals/:goalReference/update/review', [
    updateGoalController.getReviewUpdateGoalView,
  ])

  router.post('/plan/:prisonNumber/goals/:goalReference/update/review', [
    updateGoalController.submitReviewUpdateGoal,
  ])
```

(Based on the explanation above of what `prisonerSummaryRequestHandler.getPrisonerSummary` and `checkPrisonerSummaryExistsInSession` do and why/where you would use them; you could remove lines 22 and 27 altogether. I know its not part of your change, but if you want to push the broom around whilst you are in the general vicinity that would be cool 👍 . Tidy tidy 👍 )

Reply:
Removed the duplication here for both `/update/review` and `/update`

Reply:
Hmm this change has broken the test on the pipeline but its still passing locally for me 🤔

Reply:
Something strange happening with the tests here, when you move `checkPrisonerSummaryExistsInSession` to the `.use` for the Update view, the test `should be able to navigate directly to update goal page` fails 🤔

Reply:
Works fine for the Review view

Reply:
I've checked out your branch and made the edits I was suggesting to the router. Not sure what tests you had that were failing, but they are all passing now; so happy days 👍